src/main/scala/com/cognite/sdk/scala/common/dataTypes.scala

```diff
@@ -140,6 +140,8 @@ object Setter {
         case None => None
         case Some(null) => Some(SetNull()) // scalastyle:ignore null
         case Some(value: T) => Some(SetValue(value))
+        case Some(badValue) =>
+          throw new Exception(s"Expected value of type ${manifest[T]} but got '${badValue}' of type ${badValue.getClass}")
```

Comment:
This should probably at least be as `RuntimeException`, but `IllegalArgumentException` seems even more appropriate.

Try to avoid `'` as it doesn't print nicely in Databricks and possibly other places. For example, `got a value of type ${badValue.getClass}: ${badValue}` instead.

Reply:
:thinking_face: I really want to somehow quote the unexpected value, since is might be an empty string or an string containing punctuation.

Reply:
Do you know how to quote it so that it does not break Databricks?

Reply:
The usual quote characters get HTML escaped, so they print as `&quot;asd` and such. Since we're not trying to escape the bad value there will always be edge cases where the error message won't be ideal, but I wouldn't worry too much about those.

I think putting the value after a comma is clear enough. You have a good point about the empty string, and maybe null values. We can print null as `null` or `NULL`, not sure about the empty string but `got a value of type String: an empty string` wouldn't be so bad?